bluepyemodel/icselector/modules/model_selector.py

```diff
@@ -0,0 +1,156 @@
+    """selects NEURON mechanisms for cell model configuration"""
+
+    def __init__(self, ic_map):
+        """
+        Args:
+            ic_map (dict): content of the icmapping.json file
+        """
+
+        self._ic_map = ic_map
+        self._mechanisms = self._load_mechanisms()
+        self.status = "stable"
+        self.mode = "mixed"
+
+    def _get_ic_map_entry(self, key):
+        """Get an entry from the icmap
+
+        Args:
+            key (str): name of the field
+
+        Returns:
+            entry: content of the field
+        """
+
+        entry = self._ic_map[key]
+        entry = {k: v for k, v in entry.items() if not k[0] == "_"}
+        return entry
+
+    def _load_mechanisms(self):
+        """Load mechanism field from the icmap
+
+        Returns:
+            mechanisms (dict [Mechanism]):
+        """
+        ic_map_mechs = self._get_ic_map_entry("mechanisms")
+        mechanisms = {k: Mechanism(**v) for k, v in ic_map_mechs.items()}
+        return mechanisms
+
+    def _set_mechanism_parameters(self):
+        """Set mechanism parameters from icmap settings."""
+
+        mechs = self._mechanisms
+        params = self._get_ic_map_entry("mechanism_parameters")
+        for name, info in params.items():
+            mech = mechs[name]
+            mech.set_from_icmap(info)
+
+    def _get_mech_from_channel(self, channel_name):
+        """Get mechanism corresponding to channel
+
+        Args:
+            channel_name (str): name of the ion channel
+
+        Returns:
+            mech (Mechanism): corresponding mechanism
+        """
+
+        chan_to_model = self._get_ic_map_entry("channels")
+        models = chan_to_model.pop(channel_name, [])
+        mechs = [self._get_mech_from_model(model) for model in models]
+
+        for mech in mechs:
+            if self._check_mech(mech):
+                return mech
+        return None
+
+    def _check_mech(self, mech):
+        if not mech:
+            return False
+
+        if hasattr(mech, "type"):
+            return self.mode in (mech.type, "mixed")
+        return True
+
+    def _get_mech_from_model(self, model_name):
+        """Get mechanism by model name
+
+        Args:
+            model_name (str): model name
+
+        Returns:
+            (Mechanism): corresponding mechanism
+        """
+
+        if model_name in self._mechanisms:
+            mech = self._mechanisms[model_name]
+            if self._check_mech(mech):
+                return mech
+        return None
+
+    def get(self, name):
+        """Get mechanism from channel or model name
+        Args:
+            name (str): channel or model name
+        Returns:
+            (Mechanism): corresponding mechanism
+        """
+
+        mech = self._get_mech_from_channel(name)
+        if not mech:
+            mech = self._get_mech_from_model(name)
+        return mech
+
+    def select(self, name):
+        """Select an return mechanism by channel or model name
```

Comment:
typo: an -> and

Reply:
Done in [942a287](https://github.com/BlueBrain/BluePyEModel/pull/144/commits/942a287eb332cb58b5608511616eefcd2e76e880)